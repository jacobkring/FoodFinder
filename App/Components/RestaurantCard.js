// @flow
import React, { Component } from 'react'
import { TouchableOpacity, Text, Image, View } from 'react-native'
import styles from './Styles/RestaurantCardStyle'

import { Card } from 'react-native-paper';

import StarRating from './StarRating';

const priceRange = ['$', '$$', '$$$', '$$$$', '$$$$$'];

type Props = {
  onPress: Function,
  restaurant: Object
}

export default class RestaurantCard extends Component<Props> {
  render () {
    const { restaurant } = this.props.restaurant
    return (
      <TouchableOpacity style={styles.container} onPress={this.props.onPress}>
        <Card style={styles.cardContent}>
          <View style={styles.contentContainer}>
            <Image
              style={styles.restaurantThumb}
              source={{uri: restaurant.thumb}}
            />
            <View style={styles.restaurantInformationContainer}>
              <Text style={styles.restaurantNameText}>{restaurant.name}</Text>
              <View style={styles.costRatingsContainer}>
                <StarRating rating={restaurant.user_rating.aggregate_rating} />
                <Text style={styles.cuisineText}>{restaurant.user_rating.rating_text} </Text>
                <Text style={styles.cuisineText}>{priceRange[restaurant.price_range]}</Text>
              </View>
              <Text style={styles.cuisineText}>{restaurant.cuisines}</Text>
            </View>
          </View>
          <Text style={styles.cuisineText}>{restaurant.location.address}</Text>
        </Card>
      </TouchableOpacity>
    )
  }
}
