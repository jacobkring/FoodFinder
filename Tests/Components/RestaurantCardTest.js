import 'react-native'
import React from 'react'
import RestaurantCard from '../../App/Components/RestaurantCard'
import { shallow } from 'enzyme'
import renderer from 'react-test-renderer'
import PropTypes from "prop-types";

test('RestaurantCard component renders correctly', () => {
  const tree = renderer.create(
    <RestaurantCard
      restaurant={{
        restaurant: {
          name: 'AWESOME RESTAURANT',
          cuisines: 'BEST CUISINES',
          user_rating: {
            aggregate_rating: 4.2,
            rating_text: 'Very Good'
          },
          location: {
            address: 'TOP PLACE'
          },
          price_range: 1,
          thumb: 'https://b.zmtcdn.com/data/res_imagery/16777320_RESTAURANT_5bc92fcf35438e2404eb918270b2ea2c.jpg'
        }
      }}
      onPress={() => {}} />).toJSON()
  expect(tree).toMatchSnapshot()
})

test('onPress', () => {
  let i = 0 // i guess i could have used sinon here too... less is more i guess
  const onPress = () => i++
  const wrapperPress = shallow(<RestaurantCard restaurant={{
    restaurant: {
      name: 'AWESOME RESTAURANT',
      cuisines: 'BEST CUISINES',
      user_rating: {
        aggregate_rating: 4.2,
        rating_text: 'Very Good'
      },
      location: {
        address: 'TOP PLACE'
      },
      price_range: 1,
      thumb: 'https://b.zmtcdn.com/data/res_imagery/16777320_RESTAURANT_5bc92fcf35438e2404eb918270b2ea2c.jpg'
    }
  }} onPress={onPress} />)

  expect(wrapperPress.prop('onPress')).toBe(onPress) // uses the right handler
  expect(i).toBe(0)
  wrapperPress.simulate('press')
  expect(i).toBe(1)
})
