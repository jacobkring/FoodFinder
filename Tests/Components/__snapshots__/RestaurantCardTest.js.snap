// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RestaurantCard component renders correctly 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "flex": 1,
      "marginBottom": 10,
      "opacity": 1,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#ffffff",
        "borderRadius": 4,
        "elevation": 1,
        "padding": 5,
        "shadowColor": "#000000",
        "shadowOffset": Object {
          "height": 0.5,
          "width": 0,
        },
        "shadowOpacity": 0.24,
        "shadowRadius": 0.75,
      }
    }
  >
    <View
      accessible={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "flexGrow": 1,
        }
      }
    >
      <View
        index={0}
        siblings={
          Array [
            "View",
            "Text",
          ]
        }
        style={
          Object {
            "flexDirection": "row",
          }
        }
        total={2}
      >
        <Image
          source={
            Object {
              "uri": "https://b.zmtcdn.com/data/res_imagery/16777320_RESTAURANT_5bc92fcf35438e2404eb918270b2ea2c.jpg",
            }
          }
          style={
            Object {
              "height": 65,
              "width": 65,
            }
          }
        />
        <View
          style={
            Object {
              "flex": 1,
              "marginLeft": 5,
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 16,
              }
            }
          >
            AWESOME RESTAURANT
          </Text>
          <View
            style={
              Object {
                "alignItems": "center",
                "flex": 1,
                "flexDirection": "row",
                "justifyContent": "space-between",
                "margin": 0,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                }
              }
            >
              <Text
                allowFontScaling={false}
                style={
                  Array [
                    Object {
                      "color": undefined,
                      "fontSize": 12,
                    },
                    Object {
                      "color": "#ffd700",
                    },
                    Object {
                      "fontFamily": "FontAwesome",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    Object {},
                  ]
                }
              >
                
              </Text>
              <Text
                allowFontScaling={false}
                style={
                  Array [
                    Object {
                      "color": undefined,
                      "fontSize": 12,
                    },
                    Object {
                      "color": "#ffd700",
                    },
                    Object {
                      "fontFamily": "FontAwesome",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    Object {},
                  ]
                }
              >
                
              </Text>
              <Text
                allowFontScaling={false}
                style={
                  Array [
                    Object {
                      "color": undefined,
                      "fontSize": 12,
                    },
                    Object {
                      "color": "#ffd700",
                    },
                    Object {
                      "fontFamily": "FontAwesome",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    Object {},
                  ]
                }
              >
                
              </Text>
              <Text
                allowFontScaling={false}
                style={
                  Array [
                    Object {
                      "color": undefined,
                      "fontSize": 12,
                    },
                    Object {
                      "color": "#ffd700",
                    },
                    Object {
                      "fontFamily": "FontAwesome",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    Object {},
                  ]
                }
              >
                
              </Text>
              <Text
                allowFontScaling={false}
                style={
                  Array [
                    Object {
                      "color": undefined,
                      "fontSize": 12,
                    },
                    Object {
                      "color": "rgb(215,215,215)",
                    },
                    Object {
                      "fontFamily": "FontAwesome",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    Object {},
                  ]
                }
              >
                
              </Text>
            </View>
            <Text
              style={
                Object {
                  "color": "rgb(128,128,128)",
                  "fontSize": 12,
                }
              }
            >
              Very Good
               
            </Text>
            <Text
              style={
                Object {
                  "color": "rgb(128,128,128)",
                  "fontSize": 12,
                }
              }
            >
              $$
            </Text>
          </View>
          <Text
            style={
              Object {
                "color": "rgb(128,128,128)",
                "fontSize": 12,
              }
            }
          >
            BEST CUISINES
          </Text>
        </View>
      </View>
      <Text
        index={1}
        siblings={
          Array [
            "View",
            "Text",
          ]
        }
        style={
          Object {
            "color": "rgb(128,128,128)",
            "fontSize": 12,
          }
        }
        total={2}
      >
        TOP PLACE
      </Text>
    </View>
  </View>
</View>
`;
